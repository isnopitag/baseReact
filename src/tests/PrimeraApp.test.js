import React from 'react';
import '@testing-library/jest-dom'
import { render } from "@testing-library/react";
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas en PrimeraApp', () => {
    // test('Debe regresar hola soy goku', () => {
    //     const saludo = 'Hola, soy Goku';

    //     const { getByText} = render(<PrimeraApp saludo = { saludo}/>);

    //     expect( getByText(saludo) ).toBeInTheDocument();
        
    // })

    test('debe de mostrar <PrimeraApp/>', () => {
        const saludo = 'hola soy goku'
        const wrapper = shallow( <PrimeraApp saludo={saludo}/>)

        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'hola soy goku'
        const subtitulo = 'soy un subtitulo'

        const wrapper = shallow( <PrimeraApp saludo={saludo} subtitulo={subtitulo}/>)

        const textParrafo = wrapper.find('p').text();
        
        expect(textParrafo).toBe(subtitulo);

    })
    
    
    
})
