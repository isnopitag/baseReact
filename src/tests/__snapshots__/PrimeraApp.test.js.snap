// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe de mostrar <PrimeraApp/> 1`] = `
<Fragment>
  <h1>
    hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
